cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AppFabricServiceManager.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.distributed;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.twill.AbstractDistributedMasterServiceManager;
+import co.cask.cdap.proto.Containers;
+import co.cask.cdap.proto.SystemServiceLiveInfo;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.twill.api.TwillRunnerService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+import java.net.InetAddress;
+
+/**
+ * App Fabric Service Management in Distributed Mode.
+ */
+public class AppFabricServiceManager extends AbstractDistributedMasterServiceManager {
+
+  private final InetAddress hostname;
+
+  @Inject
+  public AppFabricServiceManager(CConfiguration cConf, TwillRunnerService twillRunnerService,
```

Comment:
you can implement ``MasterServiceManager`` and skip injecting unnecessary objects like ``TwillRunnerService``, etc.